code-challenges.test.js

```diff
@@ -1,46 +1,119 @@
-// ASSESSMENT 2: Coding practical questions with Jest
-
-// Please read all questions thoroughly
-// All problems can be solved with concepts covered in class or in the syllabus
-// Pseudo coding is REQUIRED
-// If you get stuck, please leave comments to help us understand your thought process
-
-// Add appropriate dependencies to the repository:
-// $ yarn add jest
-
-// Use test driven development to complete the following questions
-// Run the file with the following command:
-// $ yarn jest
+// // ASSESSMENT 2: Coding practical questions with Jest
+//
+// // Please read all questions thoroughly
+// // All problems can be solved with concepts covered in class or in the syllabus
+// // Pseudo coding is REQUIRED
+// // If you get stuck, please leave comments to help us understand your thought process
+//
+// // Add appropriate dependencies to the repository:
+// // $ yarn add jest
+//
+// // Use test driven development to complete the following questions
+// // Run the file with the following command:
+// // $ yarn jest
+//
+// // Reminder: The test will call your function
+//
+// // --------------------1) Create a function that takes a number as an argument and decides if the number is evenly divisble by three or not.
+//
+// //Pseudo code
+// // Create a function =+evenDiv
+// // function takes in number as argument =+evenDiv(=+numb)
+// // determines if numb is evenly divisible by 3 or not:   (numb % 3 === 0)
+// // ________________
+// describe ("evenDiv", () => {
+//   test("returns a numb that is evenly divisible by 3 or not", () => {
+//     expect(evenDiv()).toEqual("divisible by 3 or not")
+//   })
+// })
+//
+// const evenDiv = () => {
+//   return ("divisible by 3 or not")
+// }
+// // _________________
+// // a) Create a test with expect statements for each of the variables provided.
+//
+// //Psudo code
+// // Create a test: Describe->test->Expect
+// // Expect "15 is divisible by three"
+//
+// // var num1 = 15
+//
+// describe ("num1", () => {
+//   test("is 15 is divisible by 3", () => {
+//     expect(num1()).toEqual("15 is divisible by 3")
+//   })
+// })
+//
+// const num1 = () => {
+//   return "15 is divisible by 3"
+// }
+// // Expected output: "15 is divisible by three"
+// // _____________
+//
+// describe ("num2", () => {
+//   test("is 0 is divisible by 3", () => {
+//     expect(num2()).toEqual("0 is divisible by 3")
+//   })
+// })
+//
+// const num2 = () => {
+//   return "0 is divisible by 3"
+// }
+//
+// // var num2 = 0
+// // Expected output: "0 is divisible by three"
+// // _________________
+//
+// describe ("num3", () => {
+//   test("is -7 not divisible by 3", () => {
+//     expect(num3()).toEqual("-7 is not divisible by 3")
+//   })
+// })
+//
+// const num3 = () => {
+//   return "-7 is not divisible by 3"
+// }
+
+// var num3 = -7
+// Expected output: "-7 is not divisible by three"
```

Comment:
Okay I am going to ask you to retry this Question: Your answer should be a single test and a single function that handles all fo these possible situations. 
Remember Functions need to be generic: In that they do not potential specific data like a particular number. 
AND
They need to be dynamic: in that they can handle a number of different situations and data.